src/workerd/api/node/tests/zlib-nodejs-test.js

```diff
@@ -612,6 +612,57 @@ export const testFailedInit = {
   },
 };
 
+// Tests are taken from:
+// https://github.com/nodejs/node/blob/561bc87c7607208f0d3db6dcd9231efeb48cfe2f/test/parallel/test-zlib-destroy.js
+const zlibDestroyTest = {
+  async test() {
+    const promises = [];
+    // Verify that the zlib transform does clean up
+    // the handle when calling destroy.
+    {
+      const ts = zlib.createGzip();
+      ts.destroy();
+      assert.strictEqual(ts._handle, null);
+
+      promises.push(
+        new Promise((resolve) => {
```

Comment:
Keep in mind that you can also use `Promise.withResolvers()` to get the promise and resolve function.